File: reactapp/src/models/Query/queryModel.ts
import { TableSortDirections } from "components/Table/types/tableTypes";

export interface SortingModel<T> {
  sortBy?: keyof T;
  sortDirection?: TableSortDirections;
}

export interface PagingModel {
  pageSize: number;
  pageIndex: number;
}

export interface SearchingModel {
  searchTerm?: string;
}

export interface QueryModel<T> extends SortingModel<T>, PagingModel, SearchingModel {

}

File: WebAPI/Models/Query/QueryModel.cs
namespace SoftKiwiFlorist.Models.Query;

public enum SortDirection
{
    ASC,
    DESC,
}

public class QueryModel 
{
    public string? SearchTerm { get; set; }
    public int? PageIndex { get; set; }
    public int? PageSize { get; set; }
    public string? SortBy { get; set; }
    public SortDirection? SortDirection { get; set; }
}